lib/spack/spack/build_systems/ruby.py

```diff
@@ -0,0 +1,57 @@
+##############################################################################
+# Copyright (c) 2013-2016, Lawrence Livermore National Security, LLC.
+# Produced at the Lawrence Livermore National Laboratory.
+#
+# This file is part of Spack.
+# Created by Todd Gamblin, tgamblin@llnl.gov, All rights reserved.
+# LLNL-CODE-647188
+#
+# For details, see https://github.com/llnl/spack
+# Please also see the LICENSE file for our notice and the LGPL.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License (as
+# published by the Free Software Foundation) version 2.1, February 1999.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the IMPLIED WARRANTY OF
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the terms and
+# conditions of the GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
+##############################################################################
+
+import inspect
+
+from spack.directives import extends
+from spack.package import PackageBase, run_after
+
+
+class RubyPackage(PackageBase):
+    """Specialized class for packages that are built using Ruby
+
+    This class provides a single phase that can be overridden:
+
+    1. :py:meth:`~.RubyPackage.install`
+
+    It has sensible defaults, and for many packages the only thing
+    necessary will be to add dependencies
+    """
+    phases = ['install']
+
+    #: This attribute is used in UI queries that need to know the build
+    #: system base class
+    build_system_class = 'RubyPackage'
+
+    extends('ruby')
+
+    def install(self, spec, prefix):
+        """Install the Ruby package."""
+        inspect.getmodule(self).ruby(
```

Comment:
Why's the `getmodule` needed here?  I think you could just do `which('ruby')` (see `spack.util.executable`).  That, or, since you know the package always `depends_on('ruby')`, use `spec['ruby'].ruby`.  See @adamjstewart's #3367 for more on how that is done for the Python package.